api/py/ai/chronon/repo/compile.py

```diff
@@ -110,41 +110,65 @@ def extract_and_convert(chronon_root, input_path, output_root, debug, force_over
         if _write_obj(full_output_root, validator, name, obj, log_level, force_overwrite, force_overwrite):
             num_written_objs += 1
 
-            if obj_class is Join and feature_display:
-                _print_features_names("Output Join Features", get_join_output_columns(obj))
+            if feature_display:
+                _print_features(obj, obj_class)
 
-            if obj_class is GroupBy and feature_display:
-                _print_features_names("Output GroupBy Features", get_group_by_output_columns(obj))
-
-            # In case of online join, we need to materialize the underlying online group_bys.
-            if obj_class is Join and obj.metaData.online:
+            # In case of join, we need to materialize the following underlying group_bys
+            # 1. group_bys whose online param is set
+            # 2. group_bys whose backfill_start_date (BSD) param is set.
+            if obj_class is Join:
                 online_group_bys = {}
+                bsd_group_bys = {}
```

Comment:
nit: `offline_gb_backfilled_enabled_group_bys` 

i think the importance here is to mention that online ones are already included in `online_group_bys`